src/input.cpp

```diff
@@ -419,6 +419,16 @@ const std::vector<input_event> &input_manager::get_input_for_action( const std::
     return attributes.input_events;
 }
 
+char input_manager::get_first_char_for_action( const std::string &action_descriptor,
+        const std::string context )
+{
+    const input_event first_input_event = get_input_for_action( action_descriptor, context )[0];
+    long first_input = first_input_event.get_first_input();
+    char first_input_char = ( char )first_input;
```

Comment:
Why are you explicitly casting this using a C-style cast (there is already an implicit conversion for the assignment)?

Why the local variable, why not return it directly?

Why return a `char`, when the return value (`first_input`) is a `long` *and* the result of the function is promoted to a `long` again (see return statement in `sdl_keysym_to_curse`, which returns a `long`)?

Reply:
That is true. I've simplified new function to remove redundant type cast.